src/test/OptionSettlement.t.sol

```diff
@@ -1184,6 +1184,181 @@ contract OptionSettlementTest is Test, NFTreceiver {
         _assertTokenIsClaim(claimId);
     }
 
+    struct FuzzMetadata {
+        uint256 claimsLength;
+        uint256 totalWritten;
+        uint256 totalExercised;
+    }
+
+    function testFuzzWriteExerciseRedeem(uint32 seed) public {
+        uint32 i = 0;
+        uint256[] memory claimIds1 = new uint256[](30);
+        FuzzMetadata memory opt1 = FuzzMetadata(0, 0, 0);
+        uint256[] memory claimIds2 = new uint256[](90);
+        FuzzMetadata memory opt2 = FuzzMetadata(0, 0, 0);
+
+        // create monthly option
+        (uint256 optionId1M, IOptionSettlementEngine.Option memory option1M) = _newOption(
+            WETH_A, // underlyingAsset
+            testExerciseTimestamp, // exerciseTimestamp
+            uint40(block.timestamp + 30 days), // expiryTimestamp
+            DAI_A, // exerciseAsset
+            testUnderlyingAmount, // underlyingAmount
+            testExerciseAmount // exerciseAmount
+        );
+
+        // create quarterly option
+        (uint256 optionId3M, IOptionSettlementEngine.Option memory option3M) = _newOption(
+            WETH_A, // underlyingAsset
+            testExerciseTimestamp, // exerciseTimestamp
+            uint40(block.timestamp + 90 days), // expiryTimestamp
+            DAI_A, // exerciseAsset
+            testUnderlyingAmount, // underlyingAmount
+            testExerciseAmount // exerciseAmount
+        );
+
+        for (i = 0; i < 90; i++) {
+            // loop until expiry
+            unchecked {
+                _writeExerciseOptions(seed, option1M, optionId1M, claimIds1, opt1);
+                seed += 10;
```

Comment:
Why 10?

Reply:
completely arbitrary amount, since writeExercise increments seed a variable number of times.  I'd return the resulting value but I started running into stack too deep issues. so just went with this hacky soln.